File: frontend/src/types/todoItem.ts
export type TodoCategory =
    | "TravelAndTransport"
    | "Accommodation"
    | "DocumentsAndMoney"
    | "BookingsAndActivities"
    | "Health"
    | "Connectivity"
    | "BeforeLeaving"
    | "Other";

export type TodoItem = {
    id: string;
    tripId: string;
    name: string;
    category: TodoCategory | null;
    deadline: string | null;
    isCompleted: boolean;
    sortOrder: number;
    createdAtUtc: string;
};

export type TodoItemForm = {
    name: string;
    category: "" | TodoCategory;
    deadline: string;
};

/** Returns a fresh form object for a new to-do task. */
export const createEmptyTodoItemForm = (): TodoItemForm => ({
    name: "",
    category: "",
    deadline: "",
});

export const todoCategories: Array<{
    value: TodoCategory;
    label: string;
}> = [
    { value: "TravelAndTransport", label: "Travel & transport" },
    { value: "Accommodation", label: "Accommodation" },
    { value: "DocumentsAndMoney", label: "Documents & money" },
    { value: "BookingsAndActivities", label: "Bookings & activities" },
    { value: "Health", label: "Health" },
    { value: "Connectivity", label: "Connectivity" },
    { value: "BeforeLeaving", label: "Before leaving" },
    { value: "Other", label: "Other" },
];

File: backend/Models/TodoItem.cs
namespace TravelAssistant.Models;

/// <summary>
/// A manually managed task in one trip's to-do checklist.
/// </summary>
public class TodoItem
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid TripId { get; set; }

    public Trip Trip { get; set; } = null!;

    public string Name { get; set; } = string.Empty;

    public TodoCategory? Category { get; set; }

    /// <summary>
    /// The optional date by which the task should be completed.
    /// </summary>
    public DateOnly? Deadline { get; set; }

    public bool IsCompleted { get; set; }

    /// <summary>
    /// Stores the user's manual checklist order independently of completion state.
    /// </summary>
    public int SortOrder { get; set; }

    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;
}

/// <summary>Groups to-do tasks into the fixed categories available in the first release.</summary>
public enum TodoCategory
{
    TravelAndTransport,
    Accommodation,
    DocumentsAndMoney,
    BookingsAndActivities,
    Health,
    Connectivity,
    BeforeLeaving,
    Other
}
